smart - backends/rmp/metadata.py: Fix incorrect call to the match function

The match function should take three parameters, name, comparison, version...
The original code was passing it a reference to the object holding the data
instead, which caused the comparison in match to always fail.

Upstream-Status: Pending

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

--- a/smart/backends/rpm/metadata.py
+++ b/smart/backends/rpm/metadata.py
@@ -332,13 +332,13 @@
                     reqargs = [x for x in reqdict
                                if not ((x[2] is None or "=" in x[2]) and
                                        (RPMProvides, x[1], x[3]) in prvdict or
-                                       system_provides.match(*x[:3]))]
+                                       system_provides.match(x[1], x[2], x[3]))]
                     reqargs = collapse_libc_requires(reqargs)
 
                     recargs = [x for x in recdict
                                if not ((x[2] is None or "=" in x[2]) and
                                        (RPMProvides, x[1], x[3]) in prvdict or
-                                       system_provides.match(*x[:3]))]
+                                       system_provides.match(x[1], x[2], x[3]))]
 
                     prvargs = prvdict.keys()
                     cnfargs = cnfdict.keys()
